python/cuspatial/cuspatial/tests/spatial/distance/test_pairwise_polygon_distance.py

```diff
@@ -0,0 +1,149 @@
+
+    dlhs = cuspatial.GeoSeries(polygons1)
+    drhs = cuspatial.GeoSeries(polygons2)
+
+    expect = lhs.distance(rhs)
+    got = cuspatial.pairwise_polygon_distance(dlhs, drhs)
+
+    assert_series_equal(got, cudf.Series(expect))
+
+
+@pytest.mark.parametrize(
+    "polygons1",
+    [
+        [
+            Polygon([(0, 1), (1, 0), (-1, 0), (0, 1)]),
+            Polygon([(-4, -4), (-4, -5), (-5, -5), (-5, -4), (-5, -5)]),
+        ],
+        [
+            MultiPolygon(
+                [
+                    Polygon([(0, 1), (1, 0), (-1, 0), (0, 1)]),
+                    Polygon([(0, 1), (1, 0), (0, -1), (-1, 0), (0, 1)]),
+                ]
+            ),
+            MultiPolygon(
+                [
+                    Polygon(
+                        [(-4, -4), (-4, -5), (-5, -5), (-5, -4), (-5, -5)]
+                    ),
+                    Polygon([(-2, 0), (-2, -2), (0, -2), (0, 0), (-2, 0)]),
+                ]
+            ),
+        ],
+    ],
+)
+@pytest.mark.parametrize(
+    "polygons2",
+    [
+        [
+            Polygon([(0, 1), (1, 0), (-1, 0), (0, 1)]),
+            Polygon([(-4, -4), (-4, -5), (-5, -5), (-5, -4), (-5, -5)]),
+        ],
+        [
+            MultiPolygon(
+                [
+                    Polygon([(0, 1), (1, 0), (-1, 0), (0, 1)]),
+                    Polygon([(0, 1), (1, 0), (0, -1), (-1, 0), (0, 1)]),
+                ]
+            ),
+            MultiPolygon(
+                [
+                    Polygon(
+                        [(-4, -4), (-4, -5), (-5, -5), (-5, -4), (-5, -5)]
+                    ),
+                    Polygon([(-2, 0), (-2, -2), (0, -2), (0, 0), (-2, 0)]),
+                ]
+            ),
+        ],
+    ],
+)
+def test_two_pair(polygons1, polygons2):
+    lhs = gpd.GeoSeries(polygons1)
+    rhs = gpd.GeoSeries(polygons2)
+
+    dlhs = cuspatial.GeoSeries(polygons1)
+    drhs = cuspatial.GeoSeries(polygons2)
+
+    expect = lhs.distance(rhs)
+    got = cuspatial.pairwise_polygon_distance(dlhs, drhs)
+
+    assert_series_equal(got, cudf.Series(expect))
+
+
+def test_linestring_polygon_large(polygon_generator):
+    N = 100
+    polygons1 = gpd.GeoSeries(polygon_generator(N, 20.0, 5.0))
+    polygons2 = gpd.GeoSeries(polygon_generator(N, 10.0, 3.0))
+
+    dpolygons1 = cuspatial.from_geopandas(polygons1)
+    dpolygons2 = cuspatial.from_geopandas(polygons2)
+
+    expect = polygons1.distance(polygons2)
+    got = cuspatial.pairwise_polygon_distance(dpolygons1, dpolygons2)
+
+    assert_series_equal(got, cudf.Series(expect))
+
+
+def test_point_polygon_geoboundaries(naturalearth_lowres):
+    N = 50
+
+    lhs = naturalearth_lowres.geometry[:N].reset_index(drop=True)
+    rhs = naturalearth_lowres.geometry[N : 2 * N].reset_index(drop=True)
+    expect = lhs.distance(rhs)
+    got = cuspatial.pairwise_polygon_distance(
+        cuspatial.GeoSeries(lhs), cuspatial.GeoSeries(rhs)
+    )
+    assert_series_equal(cudf.Series(expect), got)
```

Comment:
Seems like you might have a couple of simple tests here, like `self.distance(self)` and `polygon.distance(touching)` and a pair where the distance `== 1`.

Reply:
added